### ISSUE TITLE
SubApps/SubRouters do not inherit :params

### ISSUE THREAD

**Issue by zkldi** [talentlessguy/tinyhttp#242]
**Describe the bug**

Mounting a SubApp onto an App on a given path results in the path being interpreted literally - wildcards are not passed over to the route, and neither are parameters.

**To Reproduce**

```js
// subApp.js
import { App } from "@tinyhttp/app";

const app = new App();

app.get("/profile", (req, res) => res.send("hit"));

export default app;
```

```js
// server.js
import { App } from "@tinyhttp/app";
import subApp from "./subApp.js";

const app = new App();

app.use("/users/:userID", subApp);

app.get("*", (req, res) => {
    return res.send("missed");
});
```

Making a request of `GET /users/123/profile` results in missed - doesn't seem to match params.
Making a request of `GET /users/:userID/profile` results in a hit - as in the literal string ":userID".

I haven't read the source code yet but I assume mounting subapps seems to discard any custom routes and treats it as a literal string?

**Expected behavior**

I expected subApps mounted on a route to inherit the parameters setup of the route.

**Versions**

* `node`: 15.9.0
* `@tinyhttp/app`: 1.2.20

**Additional context**

Express inherits the wildcards/custom routing when mounting subrouters onto routes, but by default it does not share parameters unless {mergeParams: true} is passed as an option to the router constructor.

**Comment by talentlessguy** [comment]
do I understand it correctly that you want to have req.params in the middleware that is mounted on subapps?

e.g. this:

```js
const app = new App()

app.use((_, res) => res.send(req.params.a))

const mainApp = new App()

mainApp.use('/:a', app)
```

**Comment by zkldi** [comment]
Yeah, that's what I want -- this could be an option in the App constructor (like expresses' `mergeParams`).

However, at the moment the bug is that the route is interpreted literally - wildcards not withstanding:

```js
mainApp.use("/:foo", app);

// GET "example.com/asdf" -> Not Found
// GET "example.com/:foo" -> found
```

**Comment by talentlessguy** [comment]
@zkldi I know why the bug occurs and it happens because for `.use` we do loose matching (that doesn't support params)

the solution would be changing a few lines in `.use` to push it as "route" instead of "mw"

**Comment by talentlessguy** [comment]
here we could add some kind of a check with `instanceof`: https://github.com/talentlessguy/tinyhttp/blob/5fe8da9247ebd0efa4078ca67f39e7743b0869f4/packages/router/src/index.ts#L121-L146

feel free to submit a bugfix PR @zkldi

**Comment by zkldi** [comment]
no problem, I'll look into it tomorrow for ya

**Comment by zkldi** [comment]
Hi, sorry. I can't wrap my head around the code at all to find the fix. Apologies.

**Comment by talentlessguy** [comment]
@zkldi no worries, will implement the fix myself then

**Comment by zkldi** [comment]
Appreciate it. Thank you for your work!

**Comment by talentlessguy** [comment]
I'm happy to announce that the bug has been fixed!

@zkldi here are passing tests for proving that:

https://github.com/talentlessguy/tinyhttp/blob/da3cdf336753b2bdf9a9cece654bcffe56c71652/__tests__/core/app.test.ts#L773-L800

currently running the build process, patch update coming soon

**Comment by talentlessguy** [comment]
fix is available in the `1.2.23` version
